Add instructions how to use status check outside if conditionals (#39464)
github/docs: content/actions/reference/workflows-and-actions/expressions.md

You can use the following status check functions as expressions in `if` conditionals. A default status check of `success()` is applied unless you include one of these functions. For more information about `if` conditionals, see [AUTOTITLE](/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif) and [AUTOTITLE](/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsif).

> None of status check functions can be used outside `if` conditionals. However, outside `if` you can use `job.status` to access same information. For more information, see [AUTOTITLE](actions/reference/contexts-reference#job-context).

Reviewer: I deleted the first sentence since it's just repeating what's in the paragraph above it.

```suggestion
Outside `if` conditionals, you can use `job.status` to access the job status. For more information, see [AUTOTITLE](actions/reference/contexts-reference#job-context).
```

Reviewer 2: You can commit suggestions from the team, but I got it for you. 👍
